### PR TITLE
Remove reference to default credentials

### LINKED ISSUE #222: [v2.12.0] Ensure CI/documentation reflect changes to default admin credentials
## Background
Previously, when installing the security plugin demo configuration, the cluster was spun up with the default admin credentials, admin:admin. A change was made in main and backported to 2.x for the 2.12.0 release, which now requires an initial admin password to be passed in via the environment variable OPENSEARCH_INITIAL_ADMIN_PASSWORD. This will break some CI/testing that relies on OpenSearch to come up without setting this environment variable. This tracking issue is to ensure compliance with the new changes.

Coming from: https://github.com/opensearch-project/security/issues/3624

## Acceptance Criteria

- [ ]  All documentation references to the old default credentials admin:admin are removed
- [ ]  Ensure that CI/testing is working with main and 2.x branches

### PR DESCRIPTION
**Pull request by derek-ho**
### Description
Removes references to default admin credentials, which were removed in 2.12

### Issues Resolved
Closes #222 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by dblock** [comment]
Let's add a CHANGELOG line?

**Comment by derek-ho** [comment]
@dblock wondering what your thoughts are here - until OpenSearch's latest image is 2.12.0, the CI will be working fine, but as soon as it is released it will fail. I was looking into adding logic into the bash script running OpenSearch is < 2.12.0 it will spin up with `https://admin:admin@os1:$PORT`, else spin up with `https://admin:myStrongPassword123!@os1:$PORT`, but that logic will break when using the `latest` tag. Do you think it's appropriate to simply leave it as is, until 2.12 is released?

**Comment by derek-ho** [comment]
Update: added a changelog and changed the PR according to my thought process - we should keep this PR open but not merged until 2.12.0 release, at which point it should be merged. I left a comment explaining the extra logic that I added. It is expected for the CI checks involving security to fail until the latest docker image is 2.12.0. @dblock let me know if this approach makes sense.

**Comment by dblock** [comment]
@derek-ho Agreed 100%, keep this around until we actually need it in 2.12. Turning this into a draft.

**Comment by DarshitChanpura** [comment]
Now that 2.12 is released this should be unblocked. Would the maintainers please review this?

**Comment by dblock** [comment]
@DarshitChanpura rebase to re-kick CI, and mark it ready for review?

**Comment by DarshitChanpura** [comment]
@derek-ho Would you mark this PR as ready of review after rebasing with main?

File: .github/actions/opensearch/run.sh [hunk 1]
```diff
@@ -55,18 +56,35 @@ if [[ $DISABLE_SECURITY = true ]]; then
...
-    appropriate/curl \
-    --max-time 120 \
-    --retry 120 \
-    --retry-delay 1 \
-    --retry-connrefused \
-    --show-error \
-    --silent \
-    --insecure \
-    https://admin:admin@os1:$PORT
+  if [[ $CLUSTER_VERSION = 'latest' ]]; then
```
Comment (derek-ho) [inline .github/actions/opensearch/run.sh:59]: Need to change this to version logic

File: .github/workflows/main.yml [hunk 2]
```diff
@@ -80,7 +80,7 @@ jobs:
 
   test-opensearch-security:
     env:
-      TEST_OPENSEARCH_SERVER: https://admin:admin@localhost:9200
+      TEST_OPENSEARCH_SERVER: https://admin:myStrongPassword123!@localhost:9200
```
Comment (derek-ho) [inline .github/workflows/main.yml:83]: Ok to hard code this here since from now on, "latest" will be > 2.12

**Comment by derek-ho** [comment]
PR is updated, but I can't install rubocop on my local and not sure if the lint checker failure is even related to this change. @nhtruong can you help me out here?

**Comment by dblock** [comment]
It may be related, something changed in a docker image? The error is strange.

```
exec /entrypoint.sh: no such file or directory
```

**Comment by derek-ho** [comment]
> It may be related, something changed in a docker image? The error is strange. (the problem is not rubocop)
> 
> ```
> exec /entrypoint.sh: no such file or directory
> ```

Very weird. Although the artifact should be the same for the given version, I just upgraded it to the latest release and that seemed to solve the problem. I think this PR is ready for review. There is one failing check, but it seems flaky. I ran the logic through a bash editor and it should work for the version, and it passes for everything else. Can the CI be re-run?

File: .github/actions/opensearch/run.sh [hunk 3]
```diff
@@ -55,6 +56,19 @@ if [[ $DISABLE_SECURITY = true ]]; then
...
+  if [ "$CLUSTER_VERSION" == "latest" ]; then
+    CREDENTIAL="admin:myStrongPassword123!"
+  else
+    COMPARE_VERSION=`echo $OPENSEARCH_REQUIRED_VERSION $CLUSTER_VERSION | tr ' ' '\n' | sort -V | uniq | head -n 1`
+    if [ "$COMPARE_VERSION" != "$OPENSEARCH_REQUIRED_VERSION" ]; then
+      CREDENTIAL="admin:admin"
+    else
+      CREDENTIAL="admin:myStrongPassword123!"
+    fi
+  fi
```
Comment (derek-ho) [inline .github/actions/opensearch/run.sh:71]: I ported this same logic from the build repo

**Review by dblock — APPROVED** [review]

**Comment by dblock** [comment]
The 1.3.2 failure doesn't look like a flake, https://github.com/opensearch-project/opensearch-ruby/actions/runs/8067723096/job/22038855221?pr=217.

**Comment by derek-ho** [comment]
> The 1.3.2 failure doesn't look like a flake, https://github.com/opensearch-project/opensearch-ruby/actions/runs/8067723096/job/22038855221?pr=217.

Reason why I think it is flaky is because the action is working on the action run on my fork: https://github.com/derek-ho/opensearch-ruby/actions/runs/8067722368/job/22038843163, but is failing on different versions there. From the logs it seems like it is stopping at the echoing of the version, and not when it is trying to access opensearch with the wrong credentials or something. Has this repo had flakiness there in the past? I also don't want to introduce flakiness in this PR.

**Comment by dblock** [comment]
I retried and it worked. Thanks. Merged.

**Pull request merged**